Watchman.DomainModel/Messages/Queries/Handlers/GetUserMessagesQueryHandler.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Watchman.Cqrs;
+using Watchman.DomainModel.Commons.Queries.Handlers;
+using Watchman.Integrations.MongoDB;
+
+namespace Watchman.DomainModel.Messages.Queries.Handlers
+{
+    public class GetUserMessagesQueryHandler : PaginationQueryHandler, IQueryHandler<GetUserMessagesQuery, GetMessagesQueryResult>
+    {
+        private readonly GetMessagesQueryHandler getMessagesQueryHandler;
+
+        public GetUserMessagesQueryHandler(ISessionFactory sessionFactory)
+        {
+            getMessagesQueryHandler = new GetMessagesQueryHandler(sessionFactory);
+        }
+
+        public GetMessagesQueryResult Handle(GetUserMessagesQuery query)
+        {
+            return getMessagesQueryHandler.Handle(query);
```

Comment:
Czy nie lepiej tutaj ograniczyć listę do danego użytkownika?

Reply:
jest ograniczana właśnie od razu po pobraniu z bazy
`csharp
            if (query is GetUserMessagesQuery userQuery)
            {
                paginated = paginated.Where(x => x.Author.Id == userQuery.UserId);
            }
`